Importação das bibliotecas

In [30]:
pip install rapidfuzz

Note: you may need to restart the kernel to use updated packages.


In [31]:
import streamlit as st
import pandas as pd
from rapidfuzz import process
import matplotlib.pyplot as plt
import seaborn as sns


_______________________

1- IMPORTAÇÃO DO DATASET

In [32]:
df=pd.read_csv("top_1000_imdb_movies.csv")

____________________________


2- ANÁLISE DE DADOS

In [33]:
###Vizualização dos dados
print("--- Primeiras 5 linhas ---")
display(df.head())

--- Primeiras 5 linhas ---


,Unnamed: 0,Movie Name,Year of Release,Watch Time,Movie Rating,Description
0,0,The Shawshank Redemption,1994,142,9.3,"Over the course of several years, two convicts..."
1,1,The Godfather,1972,175,9.2,"Don Vito Corleone, head of a mafia family, dec..."
2,2,The Dark Knight,2008,152,9.0,When the menace known as the Joker wreaks havo...
3,3,Schindler's List,1993,195,9.0,"In German-occupied Poland during World War II,..."
4,4,12 Angry Men,1957,96,9.0,The jury in a New York City murder trial is fr...


In [34]:
print("\n--- Informação do Dataset ---")
display(df.info())


--- Informação do Dataset ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Unnamed: 0       1000 non-null   int64  
 1   Movie Name       1000 non-null   object 
 2   Year of Release  1000 non-null   object 
 3   Watch Time       1000 non-null   int64  
 4   Movie Rating     1000 non-null   float64
 5   Description      1000 non-null   object 
dtypes: float64(1), int64(2), object(3)
memory usage: 47.0+ KB


None

Ao analisar esta tabela, deteto que a linha 2 'year of realease' está classificada como um objeto quando deveria ser classificada como um número por se tratar de um ano. Também detetei que 'unnamed' é uma coluna residual.

In [35]:
print("\n--- Resumo Estatístico ---")
display(df.describe())


--- Resumo Estatístico ---


,Unnamed: 0,Watch Time,Movie Rating
count,1000.000000,1000.000000,1000.000000
mean,499.500000,124.253000,7.970200
std,288.819436,28.800355,0.275732
min,0.000000,45.000000,7.600000
25%,249.750000,103.000000,7.800000
50%,499.500000,120.000000,7.900000
75%,749.250000,139.000000,8.100000
max,999.000000,321.000000,9.300000


Ao analisar esta tabela, percebo que a média de duração dos filmes são de aproximadamente 124 minutos, sendo o mais curto com 45 minutos e o mais longo com 321 minutos, mais de 5 horas. Já nas classificações, o pior filme teve uma nota de 7,6, a média é de 7,9 e a pontuação é de 9,3.

_________________

3- PRÉ-PROCESSAMENTO DE DADOS

In [ ]:
if 'Unnamed: 0' in df.columns:
        df = df.drop(columns=['Unnamed: 0'])
        
###Remoção da coluna 'Unnamed: 0' 

In [ ]:
df['Year of Release'] = df['Year of Release'].astype(str).str.extract(r'(\d{4})')
df['Year of Release'] = pd.to_numeric(df['Year of Release'])
####Conversão da coluna 'Year of Release' para o tipo numérico

In [ ]:
total_duplicados = df.duplicated().sum()

print(f"O dataset tem {total_duplicados} linhas duplicadas.")
###Verificação de linhas duplicadas no dataset

O dataset tem 0 linhas duplicadas.


In [ ]:
print("\n--- Contagem de Valores Nulos ---")
display(df.isnull().sum())
###Verificação de valores nulos no dataset


--- Contagem de Valores Nulos ---


Movie Name         0
Year of Release    0
Watch Time         0
Movie Rating       0
Description        0
dtype: int64

In [ ]:
###Verificaão do pré processamento dos dados
print("--- Informação das Colunas ---")
display(df.info())

--- Informação das Colunas ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Movie Name       1000 non-null   object 
 1   Year of Release  1000 non-null   int64  
 2   Watch Time       1000 non-null   int64  
 3   Movie Rating     1000 non-null   float64
 4   Description      1000 non-null   object 
dtypes: float64(1), int64(2), object(2)
memory usage: 39.2+ KB


None